Apps/W1/EDocumentsConnector/app/src/SignUp/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,224 @@
+        exit(SendRequest(HttpRequestMessage, HttpResponseMessage, true));
+    end;
+
+    local procedure InitRequest(var SignUpConnectionSetup: Record SignUpConnectionSetup; var HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage)
+    begin
+        Clear(HttpRequestMessage);
+        Clear(HttpResponseMessage);
+        if not SignUpConnectionSetup.Get() then
+            Error(MissingSetupErr);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage): Boolean
+    begin
+        SendRequest(HttpRequestMessage, HttpResponseMessage, false);
+    end;
+
+    local procedure SendRequest(HttpRequestMessage: HttpRequestMessage; var HttpResponseMessage: HttpResponseMessage; RootRequest: Boolean): Boolean
+    var
+        SignUpAuth: Codeunit SignUpAuth;
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+    begin
+        HttpRequestMessage.GetHeaders(HttpHeaders);
+        if RootRequest then
+            HttpHeaders.Add('Authorization', SignUpAuth.GetRootBearerAuthText())
+        else
+            HttpHeaders.Add('Authorization', SignUpAuth.GetBearerAuthText());
+        exit(HttpClient.Send(HttpRequestMessage, HttpResponseMessage));
+    end;
+
+    local procedure PrepareRequestMsg(pHttpRequestType: Enum "Http Request Type"; Uri: Text) RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+    begin
+        RequestMessage.Method(Format(pHttpRequestType));
+        RequestMessage.SetRequestUri(Uri);
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', '*/*');
+    end;
+
+    local procedure XmlToTxt(var TempBlob: Codeunit "Temp Blob"): Text
+    var
+        XMLDOMManagement: Codeunit "XML DOM Management";
+        InStr: InStream;
+        Content: Text;
+    begin
+        TempBlob.CreateInStream(InStr, TextEncoding::UTF8);
+        XMLDOMManagement.TryGetXMLAsText(InStr, Content);
+        exit(Content);
+    end;
+
+    local procedure GetDocumentType(EDocument: Record "E-Document"): Text
+    begin
+        if EDocument.Direction = EDocument.Direction::Incoming then
+            exit('ApplicationResponse');
+
+        case EDocument."Document Type" of
+            "E-Document Type"::"Sales Invoice", "E-Document Type"::"Sales Credit Memo", "E-Document Type"::"Service Invoice", "E-Document Type"::"Service Credit Memo":
+                exit('Invoice');
+            "E-Document Type"::"Issued Finance Charge Memo", "E-Document Type"::"Issued Reminder":
+                exit('PaymentReminder');
+        end;
+    end;
+
+    local procedure GetCustomerID(EDocument: Record "E-Document"): Text[50]
+    var
+        Customer: Record Customer;
+    begin
+        if EDocument.Direction <> EDocument.Direction::Outgoing then
+            exit('');
```

Comment:
Would you not rather have a runtime error here than a silent empty return? Should this procedure only be called in sending scenarios?

Reply:
True, done